docs/source/developing-volttron/developing-agents/using-asyncio-in-agents.rst

```diff
@@ -0,0 +1,130 @@
+.. _Using-Asyncio-In-Agents:
+
+===================================
+Using Asyncio in Agents
+===================================
+
+The purpose of this section to is to show how to use Asyncio with Gevent within the Agent development framework.
+
+Before we dive into the example, we establish the following concepts:
+
+* An Event Loop is a software design pattern that handles events concurrently; it waits for and dispatches multiple events concurrently and gives the illusion of executing the events in "parallel". In Python, the `Event Loop <https://docs.python.org/3.10/library/asyncio-eventloop.html#event-loop>`_ contains a list of Tasks which controls when and how those Tasks are executed.
+
+* A `Task <https://docs.python.org/3/library/asyncio-task.html#task-object>`_ is an object that runs a `coroutine <https://docs.python.org/3/library/asyncio-task.html#coroutine>`_ (i.e. asynchronous function). In Python, coroutines are written using the 'async' keyword.
+
+* A `Greenlet <https://greenlet.readthedocs.io/en/latest/#instantiation>`_ is a "lightweight coroutine for in-process sequential concurrent programming. Greenlets can be used on their own, but they are frequently used with frameworks such as gevent to provide higher-level abstractions and asynchronous I/O.
+
+* Asyncio is a built-in Python module that allows the developer to write concurrent code.
+
+* Gevent is a "coroutine-based Python networking library that uses greenlet to provide a high-level synchronous API on top of the libev or libuv event loop".
+
+The general steps to use Asyncio within the Volttron Agent framework are the following:
+
+1. Create an async method.
+2. Create a method that instantiates an Asyncio Event Loop, adds the async method created in the previous step to the Event Loop, and then starts the Event Loop.
+3. Create a method that will spawn a Greenlet and then run the Event Loop that was created in the previous step within the Greenlet.
+4. Within the constructor of the agent, use the VOLTTRON™ Interconnect Protocol (VIP) to subscribe to the method created in the previous step.
```

Comment:
4 is wrong....This is only for the config store...which is only of relevance to the openven agent because we use it.

Reply:
One could spawn the greenlet wherever they wanted to...it's not dependent on a subscription

Reply:
This makes sense. I'll update the step.